imap_processing/ialirt/utils/grouping.py

```diff
@@ -0,0 +1,100 @@
+"""Common grouping functions for I-ALiRT instruments."""
+
+import numpy as np
+import xarray as xr
+
+
+def filter_valid_groups(grouped_data: xr.Dataset) -> xr.Dataset:
+    """
+    Filter out groups where `src_seq_ctr` diff are not 1 or -16383.
+
+    Parameters
+    ----------
+    grouped_data : xr.Dataset
+        Dataset with a "group" coordinate.
+
+    Returns
+    -------
+    filtered_data : xr.Dataset
+        Filtered dataset with only valid groups remaining.
+    """
+    valid_groups = []
+    unique_groups = np.unique(grouped_data["group"].values)
+
+    for group in unique_groups:
+        src_seq_ctr = grouped_data["src_seq_ctr"][
+            (grouped_data["group"] == group).values
+        ]
+        src_seq_ctr_diff = np.diff(src_seq_ctr)
+
+        # Accept group only if all diffs are 1 or -16383
+        if np.all(np.isin(src_seq_ctr_diff, [1, -16383])):
+            valid_groups.append(group)
```

Comment:
I think you can modulo arithmetic this wrap-around. Do you want to add a log/message about groups that were skipped here too if it was invalid?
```suggestion
        src_seq_ctr_diff = np.diff(src_seq_ctr) % 16384

        if np.all(src_seq_ctr_diff == 1):
            valid_groups.append(group)
```

Reply:
Great idea!